Movies_MVVM/Modules/MoviesListTab/ViewModel/MovieListViewModel.swift

```diff
@@ -44,14 +48,28 @@ final class MovieListViewModel: MovieListViewModelProtocol {
 
     init(
         networkService: NetworkServiceProtocol,
-        imageService: ImageServiceProtocol
+        imageService: ImageServiceProtocol,
+        dataService: DataServiceProtocol,
+        keychainService: KeychainServiceProtocol
     ) {
         self.networkService = networkService
         self.imageService = imageService
+        self.dataService = dataService
+        self.keychainService = keychainService
     }
 
     // MARK: - Public methods
 
+    func getMovies() {
+        guard let movies = dataService.readMovieData(category: currentCategoryMovies.rawValue) else { return }
+        if !movies.isEmpty {
+            self.movies = movies
+            movieListState = .success
+        } else {
+            fetchMovies()
+        }
```

Comment:
```suggestion
        guard let movies = dataService.readMovieData(category: currentCategoryMovies.rawValue) else { return }
        if !movies.isEmpty {
            self.movies = movies
            movieListState = .success
        } else {
            fetchMovies()
        }
```
Если у тебя в data.Service будут уже какие-либо данные, то обновленный (актуальный) лист фильмов будет загружаться? По этому методу кажется что нет

Reply:
Почему же не будут? в guard я проверяю наличие объектов в бд, если там что то есть я устанавливаю данные от туда если нет, то делаю запрос

Reply:
Ну там данные есть (прошлогодние) и ты их покажешь, а новые не загрузишь )) я это имел ввиду

Reply:
Понял о чем речь, аналог нотификации в реалме, тут это не предусмотрено, будет над чем поработать в будущем